### ISSUE TITLE
Publish google adapter sample project results in Error 505 - Process failure

### ISSUE THREAD

**Issue by gitguy007** [BotBuilderCommunity/botbuilder-community-dotnet#201]
I am having trouble with the Google Adapter sample project as i am able to test it in the Emulator but whenever i publish, it will give me a server error 505 (as per attached screenshot).

Appreciate any advice given.

![error 505 - process failure](https://user-images.githubusercontent.com/23579183/76153552-5b713580-6108-11ea-9bcd-acfa831196bc.png)

**Comment by garypretty** [comment]
@gitguy007 Apologies for the delay here - I have researched this and have found the issue to be the Assembly Name setting within the csproj file properties.  Right now it is set to "Google Adapter Sample".  If you change this to "GoogleAdapterSample" and deploy then the project should load correctly.

**Comment by gitguy007** [comment]
Hi Gary, thanks. I am able to publish it now. However, when i try to get a response via /api/messages, it gives me an error 405.

It is the same thing with /api/actionrequests (this was stated in the instructions to setup the webhook in DialogFlow)

Any reasons why this is happening ?

![image](https://user-images.githubusercontent.com/23579183/77963215-90d80000-730f-11ea-9487-dd87472124c0.png)

**Comment by garypretty** [comment]
Yes, it's going to either be api/actionrequests I think.

**Comment by iMicknl** [comment]
@gitguy007 error 405 is because you are doing a GET request to the endpoint, while the endpoint most likely only supports a POST request. If you would use Postman or a similar tool for testing, you will notice that your endpoint is working. (or just use it for the Google Actions)

**Comment by gitguy007** [comment]
Thanks, iMicknl .. Postman seems to be returning a response from the site where I host the app service. 

When I go to my Google console to test and invoke the webhook, it gives me the error below in the first screenshot below.

This could be due to header or authentication information that needs to be provided in the second screenshot ? 

I have already tried entering the appId and appPassword in the header but it still returns the first error below.

Appreciate any help i can get :)

1.
![image](https://user-images.githubusercontent.com/23579183/77965093-e7930900-7312-11ea-986d-54e0e41a6861.png)

2.
![image](https://user-images.githubusercontent.com/23579183/77965325-4f495400-7313-11ea-939a-a2e08f7cf34b.png)

**Comment by garypretty** [comment]
@gitguy007 Did you manage to resolve this issue?  Apologies for the delayed reply.

One thing you could check is what the WebHook Type is that you have set on the adapter and ensure it is set to DialogFlow?

**Comment by gitguy007** [comment]
Hi @garypretty , it is still returning the same error to me. May i know where to check the Webhook Type ? Could it be response mapping issue ?

Btw, the sample project deploys to "/api/messages" coz that is the one working when i connect to it via Emulator. It doesn't return any response via "/api/actionrequests".

Thanks very much for any advice and guidance.

**Comment by garypretty** [comment]
Sorry to hear you are still having this issue.

The webhook type is on the options class you pass in when you create the adapter.

There is a significantly improved / refactored Google adapter on the way to NuGet in the next couple of days, so I would suggest that if you don't have any luck with the webhook type, that you try that.  I'll be publishing much improved docs for the adapter too.

**Comment by gitguy007** [comment]
Hi garypretty, i am trying to publish the new sample that was published.

Just wanted to highlight tt can run in localhost but when publishing there are errors.

There are two issues that I had to fix before being able. Here is how i resolved it and got it working online.

1. Encountered Process failure error (502.5) - Similar to my first post in this thread, the Project Assembly name needs to be renamed to "GoogleAdapterSample" - no spaces in between.

2. After above was done, encountered another error when publishing, this time it was the error message displayed below.  

This was resolved by referencing nuget Microsoft.Windows.Compatibility and publishing again.

Error message :
An error occurred while starting the application.
FileNotFoundException: Could not load file or assembly 'System.Diagnostics.DiagnosticSource, Version=4.0.3.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The system cannot find the file specified.
System.Signature.GetSignature(Void* pCorSig, int cCorSig, RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo methodHandle, RuntimeType declaringType)

InvalidOperationException: Failed to compare two elements in the array.
System.Collections.Generic.ArraySortHelper<T>.Sort(T[] keys, int index, int length, Comparison<T> comparer)
